numpy/_core/src/multiarray/temp_elide.c

```diff
@@ -109,6 +116,41 @@ find_addr(void * addresses[], npy_intp naddr, void * addr)
     return 0;
 }
 
+static int
+check_unique_temporary(PyObject *lhs)
+{
+#if PY_VERSION_HEX >= 0x030E00A7
+    // In Python 3.14.0a7 and later, a reference count of one doesn't guarantee
+    // that an object is a unique temporary variable. We scan the top of the
+    // interpreter stack to check if the object exists as an "owned" reference
+    // in the temporary stack.
+    PyFrameObject *frame = PyEval_GetFrame();
```

Comment:
You can use `_PyEval_GetFrame` to get `_PyInterpreterFrame` directly here of the current frame without having to materialize a Python level frame object, this would also be more performant.

Reply:
It looks like `_PyEval_GetFrame` isn't exposed in the public ABI, at least as of CPython commit 70b322d:

```
dlopen(/Users/goldbaum/Documents/numpy/build-install/usr/lib/python3.14/site-packages/numpy/_core/_multiarray_umath.cpython-314-darwin.so, 0x0002): symbol not found in flat namespace '__PyEval_GetFrame'
```